.github/workflows/code-quality.yml

```diff
@@ -1,77 +1,28 @@
 name: code-quality
 
-on: [push]
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - '**.py'
+      - '**.yml'
+  pull_request:
+    paths:
+      - '**.py'
+      - '**.yml'
+  workflow_dispatch:
 
 jobs:
-  isort:
-    name: Check import sorting
-    runs-on: ubuntu-latest
-
-    steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
-        with:
-          python-version: '3.11'
-      - uses: isort/isort-action@v1
-        with:
-          requirementsFiles: requirements/dev.txt
-          sortPaths: "src"
-          configuration: '--check-only --diff'
-
-  black:
-    name: Check code formatting with black
-    runs-on: ubuntu-latest
-
-    steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
-        with:
-          python-version: '3.11'
-      - name: Install dependencies
-        run: |
-          pip install -r requirements/dev.txt
-      - name: Run black
-        run: |
-          black --check src
-
-  oas-up-to-date:
-    name: Check for unexepected OAS changes
-    runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        version: ['v1', 'v2']
-    steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
-        with:
-          python-version: '3.11'
-      - uses: actions/setup-node@v4
-        with:
-          node-version: '18'
-
-      - name: Install system packages
-        run: |
-          sudo apt-get update \
-          && sudo apt-get install -y --no-install-recommends \
-            libgdal-dev \
-            gdal-bin
-
-      - name: Install dependencies
-        run: pip install -r requirements/ci.txt
-
-      - name: Generate OAS files
-        run: ./bin/generate_schema.sh ${{ matrix.version }} openapi-${{ matrix.version }}.yaml
-        env:
-          DJANGO_SETTINGS_MODULE: objecttypes.conf.ci
-
-      - name: Check for OAS changes
-        run: |
-          diff openapi-${{ matrix.version }}.yaml src/objecttypes/api/${{ matrix.version }}/openapi.yaml
-      - name: Write failure markdown
-        if: ${{ failure() }}
-        run: |
-          echo 'Run the following command locally and commit the changes' >> $GITHUB_STEP_SUMMARY
-          echo '' >> $GITHUB_STEP_SUMMARY
-          echo '```bash' >> $GITHUB_STEP_SUMMARY
-          echo './bin/generate_schema.sh ${{ matrix.version }}' >> $GITHUB_STEP_SUMMARY
-          echo '```' >> $GITHUB_STEP_SUMMARY
+  open-api-workflow-code-quality:
+    uses: maykinmedia/open-api-workflows/.github/workflows/code-quality.yml@initial-setup
+    with:
+      apt-packages: 'libgdal-dev gdal-bin'
```

Comment:
I don't think we use gdal for Objecttypes API

Reply:
I do see that the [Dockerfile](https://github.com/maykinmedia/objecttypes-api/blob/master/Dockerfile#L50) uses it, should it be removed there too?

Reply:
Good question! It's either I've mistaken and Objecttypes API uses gdal or the docker app packages were copied from the project which uses gdal (like Objects API)
Since it's out of scope for this issue, let's leave it as it is